app/controllers/CompanyDetailsNoMatchController.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import controllers.actions._
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.CompanyDetailsNoMatchView
+
+import javax.inject.Inject
+
+class CompanyDetailsNoMatchController @Inject()(appConfig: FrontendAppConfig,
+                                                override val messagesApi: MessagesApi,
+                                                getData: DataRetrievalAction,
+                                                cc: MessagesControllerComponents,
+                                                requireData: DataRequiredAction,
+                                                view: CompanyDetailsNoMatchView)
+  extends FrontendController(cc) with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (getData andThen requireData) {
```

Comment:
Was there a reason to make this its own controller? I feel like it could just be in the CompanyDetailsController with a different name (like companyDetailsNoMatch or something)

Reply:
left separately, as you cant reverse routes on a function thats not onsubmit/ onpageload and its easier for testing

Reply:
I'm not sure what you mean by "cant reverse routes on a function thats not onsubmit/ onpageload"? But easier for testing makes sense cool